engine/packages/pegboard/src/workflows/runner2.rs

```diff
@@ -0,0 +1,1176 @@
+						namespace_id: input.namespace_id,
+						last_command_idx: last_command_idx.unwrap_or(-1),
+						prepopulate_actor_names,
+						metadata,
+					})
+					.await?;
+
+				messages.push(protocol::ToClient::ToClientInit(protocol::ToClientInit {
+					runner_id: input.runner_id.to_string(),
+					last_event_idx: init_data.last_event_idx,
+					metadata: protocol::ProtocolMetadata {
+						runner_lost_threshold: runner_lost_threshold,
+					},
+				}));
+
+				// Send missed commands
+				if !init_data.missed_commands.is_empty() {
+					messages.push(protocol::ToClient::ToClientCommands(
+						init_data.missed_commands,
+					));
+				}
+
+				if !state.draining {
+					ctx.activity(InsertDbInput {
+						runner_id: input.runner_id,
+						namespace_id: input.namespace_id,
+						name: input.name.clone(),
+						key: input.key.clone(),
+						version: input.version,
+						total_slots: input.total_slots,
+						create_ts: ctx.create_ts(),
+					})
+					.await?;
+				}
+			}
+
+			let last_event_idx = events.last().map(|event| event.index);
+
+			// NOTE: This should not be parallelized because signals should be sent in order
+			// Forward to actor workflows
+			// Process events
+			for event in &events {
+				let actor_id = crate::utils::event_actor_id(&event.inner).to_string();
+				let res = ctx
+					.signal(crate::workflows::actor::Event {
+						inner: event.inner.clone(),
+					})
+					.to_workflow::<crate::workflows::actor::Workflow>()
+					.tag("actor_id", &actor_id)
+					.graceful_not_found()
+					.send()
+					.await?;
+				if res.is_none() {
+					tracing::warn!(
+						?actor_id,
+						"actor workflow not found, likely already stopped"
+					);
+				}
+			}
+
+			// If events is not empty, insert and ack
+			if let Some(last_event_idx) = last_event_idx {
+				ctx.activity(InsertEventsInput { events }).await?;
+
+				state.last_event_idx = last_event_idx;
+
+				// Ack events in batch
+				if last_event_idx
+					> state
+						.last_event_ack_idx
+						.saturating_add(EVENT_ACK_BATCH_SIZE)
+				{
+					state.last_event_ack_idx = last_event_idx;
+
+					messages.push(protocol::ToClient::ToClientAckEvents(
+						protocol::ToClientAckEvents {
+							last_event_idx: state.last_event_ack_idx,
+						},
+					));
+				}
+			}
+
+			// Insert/ack commands
+			let batch_start_index = if ack_last_command_idx != -1 || !commands.is_empty() {
+				ctx.activity(InsertCommandsInput {
+					commands: commands.clone(),
+					ack_last_command_idx,
+				})
+				.await?
+			} else {
+				None
+			};
+
+			// If commands were present during this loop iteration, forward to runner
+			if let Some(batch_start_index) = batch_start_index {
+				messages.push(protocol::ToClient::ToClientCommands(
+					commands
+						.into_iter()
+						.enumerate()
+						.map(|(index, cmd)| protocol::CommandWrapper {
+							index: batch_start_index + index as i64,
+							inner: cmd,
+						})
+						.collect(),
+				));
+			}
+
+			if check_queue {
```

Comment:
doc this is only if there's a queue of actors. this path is not used if there is enough capacity of runners, the actor wf works independently